services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -132,43 +150,37 @@ public ProduceService(Map<String, Object> props, String name) throws Exception {
       producerProps.putAll(producerPropsOverride);
 
     _producer = (KMBaseProducer) Class.forName(producerClass).getConstructor(Properties.class).newInstance(producerProps);
-
-    _running = new AtomicBoolean(false);
-    _nextIndexPerPartition = new ConcurrentHashMap<>();
-    _executor = Executors.newScheduledThreadPool(threadsNum);
-
-    MetricConfig metricConfig = new MetricConfig().samples(60).timeWindow(1000, TimeUnit.MILLISECONDS);
-    List<MetricsReporter> reporters = new ArrayList<>();
-    reporters.add(new JmxReporter(JMX_PREFIX));
-    Metrics metrics = new Metrics(metricConfig, reporters, new SystemTime());
-    Map<String, String> tags = new HashMap<>();
-    tags.put("name", _name);
-    _sensors = new ProduceMetrics(metrics, tags);
-
+    LOG.info("Producer is initialized.");
   }
 
   @Override
   public void start() {
     if (_running.compareAndSet(false, true)) {
-      int partitionNum = _partitionNum.get();
-      for (int partition = 0; partition < partitionNum; partition++) {
-        scheduleProduceRunnable(partition);
-      }
+      scheduleProduceRunnables();
+      _checkAddPartitionsExecutor.scheduleWithFixedDelay(new HandleNewPartitions(), 40_000, 40_000, TimeUnit.MILLISECONDS);
```

Comment:
I'm not sure.  If we expect the actual value of topic-management.rebalance.interval.ms to be on the order of 10 minutes then it will be an additional 10 minutes before the new partitions are picked up by the thread.